web3/_utils/method_callers.py

```diff
@@ -0,0 +1,40 @@
+
+
+class MethodBinder:
+    def __init__(self, input_formatter, output_formatter):
+        self.input_formatter = input_formatter
+        self.output_formatter = output_formatter
+
+    def __get__(self, instance, owner):
+        return instance.method_caller(
+            instance,
+            self.input_formatter,
+            self.output_formatter)
+
+
+class PropertyBinder:
+    def __init__(self, input_formatter, output_formatter):
+        self.input_formatter = input_formatter
+        self.output_formatter = output_formatter
+
+    def __get__(self, instance, owner):
+        return instance.method_caller(
```

Comment:
I think `instance` can be `None` which will result in an error if this is accessed on a class.  Probably not a big problem but it will result in an ugly stack track.  Maybe we can add a check and raise a nicer error.